pkg/controller/event/event_gc_controller.go

```diff
@@ -0,0 +1,99 @@
+package event
+
+import (
+	"sort"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	gcCheckPeriod = 20 * time.Second
+)
+
+type GCController struct {
```

Comment:
Please name it EventGC or EventGCController. Add a godoc comment here explaining what this does.